lib/utils/flattenNestedSelectorsForRule.mjs

```diff
@@ -0,0 +1,43 @@
+import resolvedNestedSelector from 'postcss-resolve-nested-selector';
+
+import getRuleSelector from './getRuleSelector.mjs';
+import getSelectorAST from './getSelectorAST.mjs';
+import isStandardSyntaxSelector from './isStandardSyntaxSelector.mjs';
+
+/**
+ * @typedef {import('postcss-selector-parser').Selector} Selector
+ * @typedef {import('postcss-selector-parser').Root} SelectorRoot
+ * @typedef {Array<{selector: Selector, resolvedSelectors: Array<Selector>}> | undefined} FlattenedSelectors
+ * @param {import('postcss').Rule} rule
+ * @param {import('stylelint').PostcssResult} result
+ * @returns {Array<{selector: Selector, resolvedSelectors: Array<Selector>}> | undefined}
```

Comment:
[suggest] Does returning an empty array instead of `undefined` make sense?

```suggestion
 * @returns {Array<{selector: Selector, resolvedSelectors: Array<Selector>}>}
```